### PR TITLE
bpo-26506: hex() documentation: mention "%x" % int (Original patch by Manvi B)

### PR DESCRIPTION
**Pull request by sharanry**
Refs:
[http://bugs.python.org/issue26506](http://bugs.python.org/issue26506)

Original patch by Manvi B

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Doc/library/functions.rst [hunk 1]
```diff
@@ -81,9 +81,23 @@ are always available.  They are listed here in alphabetical order.
...
+      >>> bin(12)
+      '0b1100'
+      >>> bin(-10)
+      '-0b1010'
+
+   If prefix "0b" is desired or not, you can use either of the following ways. See
+   also :func:`format` for more information.
+
+      >>> num = 14
+      >>> format(num, '#b'), format(num, 'b')
```
Comment (vstinner) [inline Doc/library/functions.rst]: I suggest to remove the num variable and just write 14 directly, to simplify examples (and easier to copy/paste).

File: Doc/library/functions.rst [hunk 2]
```diff
@@ -81,9 +81,23 @@ are always available.  They are listed here in alphabetical order.
...
+   a valid Python expression. If *x* is not a Python :class:`int` object, it has
+   to define an :meth:`__index__` method that returns an integer. Some examples:
+
+      >>> bin(12)
+      '0b1100'
+      >>> bin(-10)
+      '-0b1010'
+
+   If prefix "0b" is desired or not, you can use either of the following ways. See
+   also :func:`format` for more information.
```
Comment (vstinner) [inline Doc/library/functions.rst]: I suggest to move the "See also..." sentence *after* the examples.

File: Doc/library/functions.rst [hunk 3]
```diff
@@ -635,16 +649,26 @@ are always available.  They are listed here in alphabetical order.
...
       >>> hex(255)
       '0xff'
       >>> hex(-42)
       '-0x2a'
 
-   If x is not a Python :class:`int` object, it has to define an __index__()
-   method that returns an integer.
+   If you want to convert an integer number to uppercase or lowercase hexadecimal
+   string either with prefix or not, you can use either of the following ways. See
+   also :func:`format` for more information.
```
Comment (vstinner) [inline Doc/library/functions.rst]: Ditto, move "See also ..." above.

File: Doc/library/functions.rst [hunk 4]
```diff
@@ -635,16 +649,26 @@ are always available.  They are listed here in alphabetical order.
...
+
+      >>> num=255
```
Comment (vstinner) [inline Doc/library/functions.rst]: Ditto: replace num by its value (and remove num).

**Comment by vstinner** [comment]
"Original patch by Manvi B", hum, please put this string in the commit message (use "git commit --amend" for example to edit the commit message), and include the email: manvishab77@gmail.com.

**Review by vstinner — CHANGES_REQUESTED** [review]
(Please make suggested changes.)

**Comment by Mariatta** [comment]
Regarding the commit message, the core developer who merge this PR can manually type it in after they press the "Squash and merge" button :)

File: Doc/library/functions.rst [hunk 5]
```diff
@@ -81,9 +81,25 @@ are always available.  They are listed here in alphabetical order.
...
+   Some examples:
+
+      >>> bin(12)
+      '0b1100'
+      >>> bin(-10)
+      '-0b1010'
+
+   If prefix "0b" is desired or not, you can use either of the following ways. 
+   See also :
+      func:`format` for more information.
```
Comment (vstinner) [inline Doc/library/functions.rst]: This line is strange...  I suggest to put "See also ..." sentence after the following ">>> ..." examples.

**Comment by sharanry** [comment]
@haypo I have made the changes you asked. Please review them.

File: Doc/library/functions.rst [hunk 6]
```diff
@@ -81,9 +81,26 @@ are always available.  They are listed here in alphabetical order.
...
+      '-0b1010'
+
+   If prefix "0b" is desired or not, you can use either of the following ways.
+
+      >>> format(14, '#b'), format(14, 'b')
+      ('0b1110', '1110')
+      >>> f'{14:#b}', f'{14:b}'
+      ('0b1110', '1110')
+
+   See also :
```
Comment (vstinner) [inline Doc/library/functions.rst]: I don't understand why :func:`format` is splitted in two lines. I expect an error when building the doc. Please don't split this statement into two lines.

File: Doc/library/functions.rst [hunk 7]
```diff
@@ -635,16 +652,28 @@ are always available.  They are listed here in alphabetical order.
...
       >>> hex(-42)
       '-0x2a'
 
-   If x is not a Python :class:`int` object, it has to define an __index__()
-   method that returns an integer.
+   If you want to convert an integer number to uppercase or lowercase hexadecimal
+   string either with prefix or not, you can use either of the following ways.
+   See also :
+
+      func:`format` for more information.
```
Comment (vstinner) [inline Doc/library/functions.rst]: Please move "See also .." after the example.

File: Doc/library/functions.rst [hunk 8]
```diff
@@ -878,9 +907,26 @@ are always available.  They are listed here in alphabetical order.
...
+   define an :meth:`__index__` method that returns an integer. For example:
+
+      >>> oct(10)
+      '0o12'
+      >>> oct(-56)
+      '-0o70'
+
+   If you want to convert an integer number to octal string either with prefix "0o"
+   or not, you can use either of the following ways. See also :func:`format` for
+   more information.
```
Comment (vstinner) [inline Doc/library/functions.rst]: Same here. I already asked you to do that :-(

**Review by vstinner — CHANGES_REQUESTED** [review]

File: Doc/library/functions.rst (hunk 6 above)
Comment (sharanry) [reply Doc/library/functions.rst]: I don't understand what you are saying. Could you please be more specific?

Comment (vstinner) [reply Doc/library/functions.rst]: You have ":" on line 102 and "func:`format`" on line 103. The documentation uses the Sphinx format. I don't think that your syntax is valid. Please write ":func:`format`" on the same line.

Comment (sharanry) [reply Doc/library/functions.rst]: Never mind i understood what you are saying. Thanks!

**Review by vstinner — APPROVED** [review]

**Comment by vstinner** [comment]
@Mariatta: Would you mind to take a look to confirm that the doc change is fine. I don't know if we can "merge" the two "See also ..." sentence of the hex() doc. I don't know if t's worth it. Maybe just put them in the same paragraph?

**Comment by kushaldas** [comment]
I think we should ask the original author if she wants to submit the PR.

**Comment by manvi77** [comment]
I am the original contributor. I will look into it again and can submit the PR.

**Comment by Mariatta** [comment]
> I think we should ask the original author if she wants to submit the PR.

I agree. I've just updated the Dev guide with guidance and protocol for converting the patches to PR:
https://devguide.python.org/pullrequest/#converting-an-existing-patch-from-the-b-p-o-to-github

> I am the original contributor. I will look into it again and can submit the PR.

Please do.

File: Doc/library/functions.rst [hunk 9]
```diff
@@ -635,19 +651,29 @@ are always available.  They are listed here in alphabetical order.
...
+      >>> '%#x' % 255, '%x' % 255, '%X' % 255
+      ('0xff', 'ff', 'FF')
+      >>> format(255, '#x'), format(255, 'x'), format(255, 'X')
+      ('0xff', 'ff', 'FF')
+      >>> f'{255:#x}', f'{255:x}', f'{255:X}'
+      ('0xff', 'ff', 'FF')
+
+   See also :func:`format` for more information.
+
+   See also :func:`int` for converting a hexadecimal string to an integer using a base of 16.
```
Comment (vadmium) [inline Doc/library/functions.rst:676]: _Hexadecimal_ and _base of 16_ seems redundant

File: Doc/library/functions.rst [hunk 10]
```diff
@@ -635,19 +651,29 @@ are always available.  They are listed here in alphabetical order.
...
+   that returns an integer. Some examples:
 
       >>> hex(255)
       '0xff'
       >>> hex(-42)
       '-0x2a'
 
-   If x is not a Python :class:`int` object, it has to define an __index__()
-   method that returns an integer.
+   If you want to convert an integer number to uppercase or lowercase hexadecimal
```
Comment (vadmium) [inline Doc/library/functions.rst:663]: . . . to _an_ uppercase or lowercase . . .

File: Doc/library/functions.rst (hunk 9 above)
Comment (manvi77) [inline Doc/library/functions.rst]: yes, removing it.

Comment (manvi77) [reply Doc/library/functions.rst:676]: What I meant is an integer using base of 16, for eg., int('0xff', 16). If it is redundant, I can remove it.

**Comment by Mariatta** [comment]
Thanks for the PR @sharanry 
However, I accepted PR #2525 from the original author. So I will close this.
I hope you'll continue working on another issue in the bug tracker and submit a different PR.
Thanks.

**Pull request closed without merging**